### ISSUE TITLE
Parameter output power (persistent or non-persistent)

### ISSUE THREAD

**Issue by RetoLu** [suaveolent/ha-hoymiles-wifi#54]
The output power of the inverter can be set via the openDTU with two different parameters: "permanent" (persistent) and "non-permanent" (non-persistent).
There is only one parameter, is it permanent or non-permanent. 
If the permanent parameter is selected, the power is saved permanently in the inverter's internal EEPROM memory. For technical reasons, this type of memory has a limited number of write cycles.
Would it be possible to have both parameters?
See also under: https://martinteske-blog.de/home-assistant-nulleinspeisung-balkonkraftwerk/

**Comment by Tomblin** [comment]
I was wondering about the same and tried to follow it up in the code, I ended up on a constant value 8 added to a request indicating that this is a power limit change request in 
[](https://github.com/suaveolent/hoymiles-wifi/blob/f595e64303b8d15adc48a09b1a3caa2d06988fb4/hoymiles_wifi/const.py#L44)

If I understand correctly it is derived from this [](https://github.com/henkwiedig/Hoymiles-DTU-Proto/blob/main/SetPowerLimit.py) 

But I'm none the wiser which limit is set, probably the transient one, but I don't knw

**Comment by suaveolent** [comment]
TL;DR: It sets it via the permanent parameter. As of this moment, there is no way we know of to only set it temporarily. 

Long answer:
See also this dicussion:
https://github.com/suaveolent/ha-hoymiles-wifi/issues/16

The parameters and requests I took in the `const.py` file are directly copied from the Android S-Miles apk.

Unfortuntatley I did not yet find a solution to only temporarily set the power limit.

**Comment by suaveolent** [comment]
Maybe the parameters `CMD_ACTION_LIMIT_POWER_PF`or `CMD_ACTION_LIMIT_POWER_REACTIVE` could help here, but I personally did not test it or find anything usefule in the code.

**Comment by loraerial** [comment]
I would be interested to use the Parameter Outpur power.
Do you know if there is a solution to write in non-persistent ?

**Comment by Utini2000** [comment]
I am also highly interested in this since this would enable us to dynamical balance power output of any to the hoymiles connected battery without having to interact with the battery at all.

Although I am not sure if the mentioned 30 second update limit would prevent this from working anyway?

My plan was to get my apartments current power usage via Shelly 3EM every 1-2 seconds and then tell the hoymiles via HomeAssistant to increase/decrease the power output depending on the load.

**Comment by suaveolent** [comment]
@Utini2000 Unfortunately for now, there only seems to be the one mode I have already implemented. I also have not found a different feature in the app.

For the Update interval, you can check out the experimental "Performance data mode" feature (you need to trigger it each day).

**Comment by Utini2000** [comment]
@suaveolent aw crap. But just for my understanding: The opendtu stuff for non-W HMS models can write at the non-persistant EEPROM part?

**Comment by suaveolent** [comment]
Yes , opendtu has two options. However, since the W-series communicates over a different protocol, we cannot directly apply the same strategy. If hoymiles adds such an option via the protobuf interface, we can of course also add it for this integration.

**Comment by WerWolf-HH** [comment]
@suaveolent 
>> If hoymiles adds such an option via the protobuf interface, we can of course also add it for this integration.

Are you in Contact with Hoymiles about this?
Many thanks, so far, for your all effort :)

**Comment by heinihuber** [comment]
I have already given up (sorry for that!) trying to get hoymiles support to implement this feature to their cloud protocol. They don't seem to be interested in dealing with our request, instead they reply with incomprehensible text modules and then simply close the case.

@suaveolent: Thank you very much for your efforts to bring this inverter to a reasonable use!

**Comment by antonhagg** [comment]
Are there any news on this? Thinking of buying the HMS-2000-4WB. But if I cant control the power output I will look for another brand.
